peachjam/views/causelist.py

```diff
@@ -40,8 +41,18 @@ class CauseListListView(TemplateView):
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
+        # get only court classes and courts that have causelists
+        context["court_classes"] = (
```

Comment:
This is potentially expensive, and doesn't change a lot, just like court years. It'll be good to establish a caching pattern for this.

Maybe something like:

```
# this could also be on the CourtClass object
def get_court_classes_with_causelists():
  return CourtClass.objects.filter(courts__causelists__isnull=False).values_list('pk', flat=True)

...

def get_context_data(...):
  ...
  court_classes_with_causelists = cache.get_or_set('court_classes_with_causelists', get_court_classes_with_causelists)
  # similar for court pks
  return CourtClass.objects.filter(pk__in=court_classes_with_causelists).prefetch(...)
```

Reply:
changed, praise the caching